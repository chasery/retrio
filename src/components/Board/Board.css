.Board__name {
  display: flex;
  flex-grow: 0;
  min-width: 0;
  padding: 1rem;
  border-bottom: 0.0625rem solid #ebebeb;
}
.Board__name > h3 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.Board__date {
  margin: 0;
  padding: 0.75rem 1rem;
  border-top: 0.0625rem solid #ebebeb;
}
