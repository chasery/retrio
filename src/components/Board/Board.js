import React from 'react';
import { Link } from 'react-router-dom';
import BoardThumbnail from '../BoardThumbnail/BoardThumbnail';
import './Board.css';

function Board(props) {
  const { id, name, created_at, modified_at } = props;

  return (
    <li className='Board'>
      <Link to={`/boards/${id}`}>
        <div className='Board__name'>
          <h3>{name}</h3>
        </div>
        <BoardThumbnail />
        {modified_at ? (
          <p className='Board__date'>
            Last modified {modified_at.toLocaleDateString()}
          </p>
        ) : (
          <p className='Board__date'>
            Created {created_at ? created_at.toLocaleDateString() : null}
          </p>
        )}
      </Link>
    </li>
  );
}

export default Board;
