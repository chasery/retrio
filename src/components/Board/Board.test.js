import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Board from './Board';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    id: '',
  }),
}));

describe('Board component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<Board />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
