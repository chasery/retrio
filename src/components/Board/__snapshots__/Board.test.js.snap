// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component renders without error 1`] = `
<li
  className="Board"
>
  <Link
    to="/boards/undefined"
  >
    <div
      className="Board__name"
    >
      <h3 />
    </div>
    <BoardThumbnail />
    <p
      className="Board__date"
    >
      Created 
    </p>
  </Link>
</li>
`;
